reinforcement_learning/trainer/common/actor_critic.py

```diff
@@ -0,0 +1,211 @@
+# Copyright 2018 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import numpy as np
+import tensorflow as tf
+
+
+class ActorNetwork(object):
+    """
+    Input to the network is the state, output is the action
+    under a deterministic policy.
+    """
+
+    def __init__(self,
+                 sess,
+                 state_dim,
+                 action_dim,
+                 action_high,
+                 action_low,
+                 learning_rate,
+                 grad_norm_clip,
+                 tau,
+                 batch_size,
+                 name=None):
+        self.sess = sess
+        self.s_dim = state_dim
+        self.a_dim = action_dim
+        self.a_high = action_high
+        self.a_low = action_low
+        self.learning_rate = learning_rate
+        self.grad_norm_clip = grad_norm_clip
+        self.tau = tau
+        self.batch_size = batch_size
+
+        # create networks
+        net_name = 'actor' if name is None else name
```

Comment:
is this necessary?  i would suggest removing it to reduce distraction if not.  same comment on the critic network class.

Reply:
TD3 and C2A2 has multiple critics/actors, I use net_name to create separate scopes.